edx/edx-enterprise
# Update consent API to use catalog service user; fix coverage on master

**Description:**

This pull request has two major features:

1. Switch the consent API over to use `CourseCatalogApiServiceClient` class, which populates with a token from the system catalog service user, rather than from the user for a given request. This solves for an issue where non-privileged users encountered authorization errors for the Data Sharing Consent API.

2. Fix missing coverage on the master branch

**JIRA:** No relevant JIRA

**Dependencies:** None

**Merge deadline:** ASAP

**Installation instructions:**

1. Ensure devstack is configured with course discovery
2. Ensure a catalog is created in course discovery
3. Ensure EnterpriseCustomer is linked to the given catalog

**Testing instructions:**

1. Using edx-enterprise master (not this branch), log in with a non-privileged user, and access the consent URL following the pattern:
    ```
    http://localhost:18000/consent/api/v1/data_sharing_consent?username=unpriv_user&enterprise_customer_uuid=ent-uuid-here&course_id=course-v1:edX%2BDemoX%2BDemo_Course
    ```
2. Verify that a 400-series authorization error occurs.
3. Pull this branch and restart the LMS
4. Refresh the aforementioned URL and verify that no error occurs.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.
